amazon/iontest/ion_test_driver.py

```diff
@@ -931,6 +921,18 @@ def validate_results(report, result_field, read_error, read_compare, write_error
             raise ValueError("Invalid report: result passes with error(s) for '" + impl + "' in '" + test_file + "'.")
 
 
+def replace_impl_name_for_message(obj, first_impl, second_impl):
+    if '_' not in first_impl or '_' not in second_impl or 'message' not in obj.keys():
+        return
+    obj['message'] = obj['message'].replace(first_impl, first_impl.split('_')[0])
+    obj['message'] = obj['message'].replace(second_impl, second_impl.split('_')[0])
+
+
+def replace_impl_name_for_obj(errors, first_impl, second_impl):
+    for error in errors:
+        replace_impl_name_for_message(error, first_impl, second_impl)
+
```

Comment:
Please add comments in the code to explain why these are needed. The description in the PR makes sense, but it is a good idea to have a similar explanation in the code to make it easier to follow.

Reply:
Added.